src/levanter/eval_harness.py

```diff
@@ -0,0 +1,564 @@
+    Harness expects.
+
+    nb that LM Eval Harness has its own TaskConfig, but its defaults are not the same as just passing in
+    a dict, and we want the behavior of passing in a dict.
+
+    See Also:
+        [LM Eval Harness TaskConfig](https://github.com/EleutherAI/lm-evaluation-harness/blob/0ef7548d7c3f01108e7c12900a5e5eb4b4a668f7/lm_eval/api/task.py#L55)
+    """
+
+    task: str
+    task_alias: str | None = None
+    num_fewshot: int | None = None
+
+    use_prompt: str | None = None
+    description: str | None = None
+    target_delimiter: str | None = None
+    fewshot_delimiter: str | None = None
+
+    def to_dict(self):
+        base_dict = dataclasses.asdict(self)
+        return {k: v for k, v in base_dict.items() if v is not None}
+
+
+@dataclass(frozen=True)
+class LmEvalHarnessConfig:
+    task_spec: list[TaskConfig | str] | None = None
+    max_examples: int | None = None
+    max_eval_length: int | None = None
+    log_samples: bool = False
+
+    def task_spec_or_default(self) -> list[str | dict]:
+        if self.task_spec is None:
+            return ["hellaswag", "piqa"]
+        return [task.to_dict() if isinstance(task, TaskConfig) else task for task in self.task_spec]
+
+    def to_task_dict(self) -> dict:
+        import lm_eval.tasks as tasks
+
+        manager = tasks.TaskManager()
+        # we need to do it this way b/c i can't figure out how to run e.g. hellaswag 0 shot and 10 shot in a single run
+        this_tasks = {}
+        for task in self.task_spec_or_default():
+            try:
+                if isinstance(task, str):
+                    this_tasks.update(tasks.get_task_dict(task, manager))
+                else:
+                    our_name = task.get("task_alias", task["task"]) if isinstance(task, dict) else task
+                    our_name = our_name.replace(" ", "_")
+                    task_dict = tasks.get_task_dict([task], manager)
+                    this_task = task_dict.popitem()[1]
+                    # hacky, but this allows us to run multiple instances of the same task with different fewshot settings
+                    this_task.config.task = our_name
+                    this_tasks[our_name] = this_task
+            except Exception:
+                logger.exception(f"Failed to load task {task}")
+                raise ValueError(f"Failed to load task {task}")
+        return this_tasks
+
+
+@dataclass(frozen=True)
+class EvalHarnessMainConfig:
+    tokenizer: str
+    checkpoint_path: str
+    checkpoint_is_hf: bool = False
+    trainer: TrainerConfig = dataclasses.field(default_factory=TrainerConfig)
+    model: LmConfig = dataclasses.field(default_factory=Gpt2Config)
+
+    eval_harness: LmEvalHarnessConfig = dataclasses.field(default_factory=LmEvalHarnessConfig)
+
+    @property
+    def EvalBatch(self):
+        return self.trainer.EvalBatch
+
+    @cached_property
+    def the_tokenizer(self):
```

Comment:
not obvious what this function is doing from the name...maybe `tokenizer_object` or something?

Reply:
i use this convention throughout levanter...